### PR TITLE
Use k8s-test-infra devel image

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This uses the images defined in https://github.com/NVIDIA/k8s-test-infra over per-project images.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,8 +26,9 @@ ifeq ($(IMAGE_NAME),)
 IMAGE_NAME = $(REGISTRY)/$(DRIVER_NAME)
 endif
 
-BUILDIMAGE_TAG ?= golang$(GOLANG_VERSION)
-BUILDIMAGE ?= $(IMAGE_NAME)-build:$(BUILDIMAGE_TAG)
+GO_VERSION := $(shell awk '/^go /{print $$2}' go.mod|head -n1)
```
Comment (elezar) [inline Makefile:29]: Why not use `versions.mk` as we do in our other projects?

File: Makefile [hunk 2]
```diff
@@ -26,8 +26,9 @@ ifeq ($(IMAGE_NAME),)
...
+BUILDIMAGE_TAG ?= devel-go$(GO_VERSION)
+BUILDIMAGE ?=  ghcr.io/nvidia/k8s-test-infra:$(BUILDIMAGE_TAG)
```
Comment (elezar) [inline Makefile:31]: Let's define this all in `versions.mk`.

**Review by elezar — CHANGES_REQUESTED** [review]

**Comment by elezar** [comment]
Please also see the related comments here: https://github.com/NVIDIA/k8s-test-infra/pull/3#discussion_r1466644781

**Comment by ArangoGutierrez** [comment]
> Please also see the related comments here: [NVIDIA/k8s-test-infra#3 (comment)](https://github.com/NVIDIA/k8s-test-infra/pull/3#discussion_r1466644781)

https://github.com/NVIDIA/k8s-test-infra/pull/4

**Comment by elezar** [comment]
The linked PR looks good. Please update the use of `GOLANG_VERSION`.

File: Makefile (hunk 2 above)
Comment (elezar) [inline Makefile:31]: Add `BUILDIMAGE ?= ghcr.io/nvidia/k8s-test-infra:golang$(GOLANG_VERSION)` to `versions.mk` directly.

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,4 @@
 /nvidia-dra-plugin
 .idea
 [._]*.sw[a-p]
+.coverage.*
```
Comment (elezar) [inline .gitignore:8]: nit: no trailing newline

File: Makefile [hunk 4]
```diff
@@ -138,34 +138,14 @@ generate-clientset: .remove-clientset .remove-deepcopy .remove-crds
...
 	$(DOCKER) run \
 		--rm \
-		-e HOME=$(PWD) \
-		-e GOCACHE=$(PWD)/.cache/go \
-		-e GOPATH=$(PWD)/.cache/gopath \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.gocache \
+		-v $(PWD):/work \
+	    -v $(GOMODCACHE):/go/pkg/mod \
```
Comment (elezar) [inline Makefile:147]: Nit: Update the indent to match the surrounding code.

Also, what is the impact of this if the golang version in the container is different to the host? Should this also be `/tmp/go/pkg/mod` or something?

Comment (elezar) [reply Makefile:147]: Note that the `.shell` target needs to be updated to match this target too.

File: Makefile [hunk 5]
```diff
@@ -26,8 +26,8 @@ ifeq ($(IMAGE_NAME),)
 IMAGE_NAME = $(REGISTRY)/$(DRIVER_NAME)
 endif
 
-BUILDIMAGE_TAG ?= golang$(GOLANG_VERSION)
-BUILDIMAGE ?= $(IMAGE_NAME)-build:$(BUILDIMAGE_TAG)
+GOCACHE := $(shell go env GOCACHE)
```
Comment (elezar) [inline Makefile:29]: These are no longer required?

File: Makefile [hunk 6]
```diff
@@ -138,34 +135,14 @@ generate-clientset: .remove-clientset .remove-deepcopy .remove-crds
...
+	@echo "Running 'make $(*)' in container image $(BUILDIMAGE)"
 	$(DOCKER) run \
 		--rm \
-		-e HOME=$(PWD) \
-		-e GOCACHE=$(PWD)/.cache/go \
-		-e GOPATH=$(PWD)/.cache/gopath \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.cache/go \
+		-e GOMODCACHE=/tmp/.cache/gopath \
```
Comment (elezar) [inline Makefile:143]: OK. Now I see the confusion. We no longer set `GOPATH` here. Which of these need to be set?

Comment (ArangoGutierrez) [reply Makefile:143]: GOPATH is `/go` by default in the `golang` images
```bash
root@83d456ef8ffe:/go# env
HOSTNAME=83d456ef8ffe
PWD=/go
HOME=/root
GOLANG_VERSION=1.21.6
TERM=xterm
SHLVL=1
GOTOOLCHAIN=local
PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
GOPATH=/go
_=/usr/bin/env
```

Comment (ArangoGutierrez) [reply Makefile:143]: so here we are only defining the ones we want to modify

Comment (elezar) [inline Makefile:143]:
```suggestion
		-e GOMODCACHE=/tmp/.cache/gomod \
```

File: Makefile [hunk 7]
```diff
@@ -176,10 +153,9 @@ PHONY: .shell
 	$(DOCKER) run \
 		--rm \
 		-ti \
-		-e HOME=$(PWD) \
-		-e GOCACHE=$(PWD)/.cache/go \
-		-e GOPATH=$(PWD)/.cache/gopath \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.cache/go \
+		-e GOMODCACHE=/tmp/.cache/gopath \
```
Comment (elezar) [inline Makefile:157]:
```suggestion
		-e GOMODCACHE=/tmp/.cache/gomod \
```

**Review by elezar — CHANGES_REQUESTED** [review]
One comment for clarity.

File: .github/workflows/build-images.yaml
Comment (elezar) [inline .github/workflows/build-images.yaml:1]: Let's name this `images.yaml` like in `k8s-device-plugin` and `mig-parted`.

File: .github/workflows/golang-build.yaml
Comment (elezar) [inline .github/workflows/golang-build.yaml:1]: Let's align the naming of this file with the other repos.

**Review by elezar — CHANGES_REQUESTED** [review]
Minor comments w.r.t aligning with other repos.

**Review by elezar — APPROVED** [review]
/lgtm

**Pull request merged**